src/main/scala/tech/cryptonomic/conseil/tezos/PlatformDiscoveryOperations.scala

```diff
@@ -88,4 +93,72 @@ object PlatformDiscoveryOperations {
   private def makeDisplayName(name: String): String = {
     name.capitalize.replace("_", " ")
   }
+
+
+  /** Makes list of possible string values of the attributes
+    * @param  tableName name of the table from which we extract attributes
+    * @param  attribute name of the attribute
+    * @param  withFilter optional parameter which can filter attributes
+    * @return list of attributes
+    **/
+  def listAttributeValues(tableName: String, attribute: String, withFilter: Option[String] = None)
+    (implicit ec: ExecutionContext): Future[List[String]] = {
+    val res = verifyAttributesAndGetQueries(tableName, attribute, withFilter)
+    ApiOperations.runQuerySequence(res).map(_.flatten)
+  }
+
+  /** Makes list of DBIO actions to get possible string values of the attributes
+    * @param  tableName name of the table from which we extract attributes
+    * @param  attribute name of the attribute
+    * @param  withFilter optional parameter which can filter attributes
+    * @return list of DBIO actions to get matching attributes
+    **/
+  def verifyAttributesAndGetQueries(tableName: String, attribute: String, withFilter: Option[String])
+    (implicit ec: ExecutionContext): List[DBIO[List[String]]] = {
+    for {
+      (name, table) <- tablesMap
+      if name == tableName
+      col <- table.baseTableRow.create_*
+      if col.name == attribute
+    } yield makeAttributesQuery(name, col, withFilter)
+  }
+
+  /** Makes list of possible string values of the attributes
+    * @param  tableName name of the table from which we extract attributes
+    * @param  column name of the attribute
+    * @param  withFilter optional parameter which can filter attributes
+    * @return list of attributes
+    **/
+  private def makeAttributesQuery(tableName: String, column: FieldSymbol, withFilter: Option[String])
+    (implicit ec: ExecutionContext): DBIO[List[String]] = {
+    for {
+      distinctCount <- TezosDb.countDistinct(tableName, column.name)
+      if canQueryType(mapType(column.tpe)) && !isHighCardinality(distinctCount)
+      distinctSelect <- withFilter match {
+        case Some(filter) =>
+          TezosDatabaseOperations.selectDistinctLike(tableName, column.name, sanitizeForSql(filter))
+        case None =>
+          TezosDatabaseOperations.selectDistinct(tableName, column.name)
+      }
+    } yield distinctSelect
+  }
+
+  /** Checks the data types if cannot be queried by */
+  private def canQueryType(dt: DataType): Boolean = {
+    // values described in the ticket #183
+    val cantQuery = List(DataType.Date, DataType.DateTime, DataType.Int, DataType.LargeInt, DataType.Decimal)
+    !cantQuery.contains(dt)
+  }
+
+  /** Checks if cardinality of the column is to high so it should not be queried */
+  private def isHighCardinality(distinctCount: Int): Boolean = {
```

Comment:
What about reversing the check and name this as an acceptance function (e.g. `isAcceptableColumnCardinality`, `isLimitedCardinality`, `isLowCardinality`...)?

Reply:
ok, changed